owasp-top10-2017-apps/a1/copy-n-paste/README.md

```diff
@@ -1,40 +1,166 @@
 
-## Requirements
+The main goal of this project is to discuss how **SQL Injection** vulnerabilities can be exploited and to encourage developers send Pull Requests to secDevLabs on how they would mitigate these flaws.
+
+## Setup
+
+To start this intentionally **insecure application**, you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you must type the following commands to start:
+
+```sh
+cd secDevLabs/owasp-top10-2017-apps/a1/copy-n-paste
+```
+
+```sh
+make install
+```
 
-To build this lab you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install].
+Then simply visit [localhost:3000][App], as exemplified below:
+
+<img src="images/CopyNPaste.png" align="center"/>
 
-## Deploy and Run
+## Get to know the app 💉
 
-After cloning this repository, you can type the following command to start the vulnerable application:
+To properly understand how this application works, you can follow these simple steps:
 
+- Register a new user via front-end.
+- Login as this user via front-end.
+* Register another user now using command line:
 ```sh
-$ make install
+curl -s -H "Content-Type: application/json" -d '{"user":"bob", "pass":"password", "passcheck":"password"}' http://localhost:3000/register
 ```
+* Login as this second user now using command line:
+```sh
+curl -s -H "Content-Type: application/json" -d '{"user":"bob", "pass":"password"}' http://localhost:3000/login
+
+```
+
+## Attack narrative
+
+Now that you know the purpose of this app, what could possibly go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or it's users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+### 👀
+
+#### Lack of input validation allows for remote code execution
+
+After reviewing `NewUser()` , `CheckIfUserExists()` and `AuthenticateUser()` from [`db.go`]((https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a1/copy-n-paste/app/util/db.go#)) file, it was possible to see that some input from users are concatenated with SQL queries, as shown in the following code snippets:
+
+```go
+query := fmt.Sprint("select * from Users where username = '" + user + "'")
+
+```
+
+```go
+query := fmt.Sprint("insert into Users (username, password) values ('" + user + "', '" + passHash + "')")
+```
+
+```go
+query := fmt.Sprint("select username from Users where username = '" + username + "'")
+```
+
+As no validation is being made on these variables, SQL injections may be successfully executed in the database. Using the web interface, we can send some information, using the form from "inspect page", to better understand how it communicates with the API.
```

Comment:
I would use `Developer Tools` (and add it's hotkey here) insted of `inspect page`.